recursion/recursion.md

```diff
@@ -125,6 +125,70 @@ int fun1(int x, int y)
 What do these function calls return?
 
 `fun1(1, 1)`
-`fun1(2, 1)`
+`fun1(2, 1)` 
 `fun1(2, 2)`
 `fun1(0, 2)`
+
+#### [Challenge] - GCD
+
+The [GCD] (https://en.wikipedia.org/wiki/Greatest_common_divisor)of two or more integers, which are not all zero, is the largest positive integer that divides each of the integers.
+The greatest common divisor(GCD) is also known as:-
+* the greatest common factor (gcf),
+* highest common factor (hcf),
+* greatest common measure (gcm),
+* highest common divisor.
+
+
+For example:-
+
+the gcd of 48 and 14 is 2.
+
+### pseudocode to calculate gcd of two numbers:
+
+GCD(x, y)
+Begin
+      if y = 0 then
+          return x;
+      else
+          Call: GCD(y, x%y);
+      endif
+End
+
+### Check for understanding
+
+Finding GCD of 48 & 14 :
+14 ) 48 ( 3
```

Comment:
Enclose this inside <code>```</code> to make it look like a code. Add a new line before this line.

Reply:
before which line?
 ### Check for understanding or Finding GCD of 48 & 14

Reply:
@praveenscience ☝️

Reply:
@Kumaranupam2020 Line 166 should be empty. 😄

Reply:
i think i have solved everything as requirred

Reply:
Will check them... 😇